### PR TITLE
Add a flushSync() method

### PR DESCRIPTION
**Pull request by dbkr**
Hopefully all explained in the comments / doc strings

Open to design suggestions on this one - alternatives discussed in the comments. Another option that now comes to mind is to let you supply a callback to tick the fake timers in addition to the real setTimeout function, so mock-request would tick your timers, flushing the event loop each time, until it saw the requests you wanted.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -96,6 +96,37 @@ class HttpBackend {
...
+        // Note that an alternative to this method could be to let the app pass
+        // in a setTimeout function so it could give us the real setTimeout function
+        // rather than the faked one. However, if you're running with fake timers
+        // the only thing setting a real timer would do is allow pending promises
+        // to resolve/reject. The app would have no way to know when the correct,
+        // non-racy point to tick the timers is.
+        console.log(`${Date.now()} HTTP backend flushing (sync)... (path=${path}`
+            + ` numToFlush=${numToFlush}`
+            + ")"
+        );
```
Comment (SimonBrandner) [inline src/index.ts]:
```suggestion
        console.log(
        	`${Date.now()} HTTP backend flushing (sync)... (path=${path} ` +
            `numToFlush=${numToFlush})`
        );
```

Just a personal preference :sweat_smile:

File: src/index.ts [hunk 2]
```diff
@@ -96,6 +96,37 @@ class HttpBackend {
...
+
+        let numFlushed = 0;
+        while ((!numToFlush || numFlushed < numToFlush) && this._takeFromQueue(path)) {
```
Comment (SimonBrandner) [inline src/index.ts:124]: What will happen if `numToFlush` is `0`?

Comment (SimonBrandner) [inline src/index.ts:124]: I am not completely sure I understand how the return value of `_takeFromQueue()` influences whether we should continue. What am I missing?

Comment (dbkr) [reply src/index.ts:124]: It'll treat it as unlimited - I don't think there's any reason to ask it to flush no requests, so this is probably a sensible use of zero, and is consistent with the rest of the API.

Comment (dbkr) [reply src/index.ts:124]: _takeFromQueue returns false if it didn't take anything, so it stops iterating once it reaches the end of the queue.

**Review by SimonBrandner — APPROVED** [review]
Thanks!

**Pull request merged**